src/items/menu.js

```diff
@@ -122,12 +122,17 @@ export default class MenuPanelItem extends PanelItem {
     };
 
     const onclick = (ev) => {
+      if (this.state.isOpen) {
+        this.core.make('osjs/contextmenu').hide();
+        return;
+      }
+
       const packages = this.core.make('osjs/packages')
         .getPackages(m => (!m.type || m.type === 'application'));
 
       this.core.make('osjs/contextmenu').show({
         menu: makeTree(this.core, icon, __, [].concat(packages)),
-        position: ev.target,
+        position: ev.target.closest('.osjs-panel-item--clickable'),
```

Comment:
I try to support IE11 out of the box, and `.closest()` is sadly not implemented there. I assume this is to get a bit more consistent positioning, yes ?

Looking at the `@osjs/panels` codebase, it might be better to change the implementation over there so there's a `this.$element` on each `PanelItem` instance. Because I now see that the panel items are put into the DOM just as you discovered in the contextmenu (directly on the parent element) :thinking: 

1. https://github.com/os-js/osjs-panels/blob/def3032fa92612c4f8b6987860f0846be27de854/src/panel-item.js#L79
2. https://github.com/os-js/osjs-panels/blob/def3032fa92612c4f8b6987860f0846be27de854/src/panel.js#L151

Reply:
Ok that should be an easy fix.

Reply:
Looks like there is a cleaner solution to this in Hyperapp 2.x. (Point of information)

Reply:
I'm going to plan the upgrade path to Hyperapp 2 down the line.